tests/docker/Dockerfile

```diff
@@ -262,7 +262,7 @@ COPY --from=k8s /usr/local/bin/kubectl /usr/local/bin/helm /usr/local/bin/
 COPY --from=kaf /usr/local/bin/kaf /usr/local/bin/
 COPY --from=kcl /usr/local/bin/kcl /usr/local/bin/
 COPY --from=kgo-verifier /opt/kgo-verifier /opt/kgo-verifier
-COPY --from=byoc-mock /opt/redpanda-tests/byoc-mock/.rpk.managed-byoc /root/.local/bin/.rpk.managed-byoc
+COPY --from=byoc-mock /opt/redpanda-tests/go/byoc-mock/.rpk.managed-byoc /root/.local/bin/.rpk.managed-byoc
```

Comment:
qq: does it have to be this path? `/root/.local/bin/.rpk.managed-byoc`

Reply:
Yes, this is needed for the test (https://github.com/redpanda-data/redpanda/pull/12456). The test is an `rpk plugin list --local` test which internally looks for installed plugins under (`$HOME/.local/bin/`)